### ISSUE TITLE
[BUG] Failure serializing singletonSet.

### ISSUE THREAD

**Issue by benallard** [jobrunr/jobrunr#282]
**Are you running the latest version of JobRunr?**
4.0.2

**Describe the bug**
NullPointerException when running an enqueud job.

**Environment**
I'm using JobRunr version: 4.02
I'm running on JRE / JDK (e.g. OpenJDK 1.8.0_292): 8
I'm using the following StorageProvider (e.g. Oracle / MongoDB / ...): postgresql

**How to Reproduce**
Provide a simple Github project or code fragment which shows how to reproduce the issue (ideally a Github repo).
```java
Long id = 215474;
BackgroundJob.enqueue(() -> service.helpMe(Collections.singleton(id)));
```
Here the serialized job. clearly, both the name and the signature show that a collection was there. the parameter on the other side serialized is `null`.
``` json
{
  "version": 45,
  "jobSignature": "com.acme.logic.report.MetabolicBalanceAnimalAnalysisGenerator.fromRounds(java.util.Collections$SingletonSet)",
  "jobName": "com.acme.logic.report.MetabolicBalanceAnimalAnalysisGenerator.fromRounds([215474])",
  "jobDetails": {
    "className": "com.acme.logic.report.MetabolicBalanceAnimalAnalysisGenerator",
    "staticFieldName": null,
    "methodName": "fromRounds",
    "jobParameters": [
      {
        "className": "java.util.Collection",
        "actualClassName": "java.util.Collection",
        "object": null
      }
    ],
    "cacheable": false
  },
  "id": "2d719995-c52d-4f3f-a68f-790e7afceb44",
```

I solved it by using another method, but I'm doubting I might be the only one impacted.

**Expected behavior**
Enqueue method runs.

**Additional context**

**Comment by rdehuyss** [comment]
I'm happy to accept a PR for this one

**Comment by benallard** [comment]
I actually saw the problem last week already an cloned the repository locally. I wasn't able to reproduce the problem with a small test though. I was somehow hopping it had already been solved in the meantime. I'll see if I can come closer on a second try.

**Comment by benallard** [comment]
I didn't came further with manual testing.
I can reproduce it at will with my project though.
What I found:
The job returned by `storageProvider.save()` is not the same as you get through `storageProvider.getJobById()`
the following screenshot is from inside `org.jobrunr.scheduling.AbstractJobScheduler#saveJob`. It's been properly saved, though not the way we expected.
![image](https://user-images.githubusercontent.com/178188/145683288-52d7bb67-0cfd-41a6-97fc-8e1c6facb0ce.png)

**Comment by benallard** [comment]
That really looks like a deserialization problem with Jackson.

The insert statement is correct, and the result from the select is correct as well.

We use 2.12.3
JobRunR testsuite uses 2.13.0

Seeing if I upgrading our project solves the issue. Nope. Must be some weird Jackson customization.

**Comment by benallard** [comment]
Ok, the issue is there the whole time in front of me, in my manual test I just serialized and deserialized the job, but I didn't ran it. it would have exploded on the first attempt.

Following patch is failing.

``` patch
Index: core/src/test/java/org/jobrunr/jobs/mappers/JobMapperTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/test/java/org/jobrunr/jobs/mappers/JobMapperTest.java b/core/src/test/java/org/jobrunr/jobs/mappers/JobMapperTest.java
--- a/core/src/test/java/org/jobrunr/jobs/mappers/JobMapperTest.java	(revision b7f47980a7603118a5b6b025f8d89460dfd9f724)
+++ b/core/src/test/java/org/jobrunr/jobs/mappers/JobMapperTest.java	(date 1639242948260)
@@ -88,6 +88,18 @@
         assertThatCode(() -> jobMapper.deserializeJob(jobAsString)).doesNotThrowAnyException();
     }
 
+    @Test
+    void testSerializeAndDeserializeJobWithCollection() {
+        Long value = 1L;
+        Job job = anEnqueuedJob().withJobDetails(() -> testService.doWorkWithCollection(Collections.singleton(value))).build();
+
+        String jobAsString = jobMapper.serializeJob(job);
+
+        Job deserializedJob = jobMapper.deserializeJob(jobAsString);
+        assertThat(deserializedJob.getJobDetails())
+                .hasArgs(Collections.singleton(value));
+    }
+
     @Test
     void serializeAndDeserializeRecurringJob() {
         RecurringJob recurringJob = aDefaultRecurringJob().build();

```

**Comment by benallard** [comment]
It's definitely conflicting with the fix for #254, but even when removing the fix it's not working either. I have no idea what's happening here.

**Comment by benallard** [comment]
Thank you.
